crates/core/tedge/src/system_services/managers/config.rs

```diff
@@ -0,0 +1,139 @@
+use crate::system_services::*;
+use serde::Deserialize;
+use std::fs;
+use std::path::PathBuf;
+
+pub const SERVICE_CONFIG_FILE: &str = "system.toml";
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct SystemConfig {
+    pub(crate) init: InitConfig,
+}
+
+#[derive(Deserialize, Debug, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct InitConfig {
+    pub name: String,
+    pub is_available: Vec<String>,
+    pub restart: Vec<String>,
+    pub stop: Vec<String>,
+    pub enable: Vec<String>,
+    pub disable: Vec<String>,
+    pub is_active: Vec<String>,
+}
+
+impl Default for InitConfig {
+    fn default() -> Self {
+        Self {
+            name: "systemd".to_string(),
+            is_available: vec!["/bin/systemctl".into(), "--version".into()],
+            restart: vec!["/bin/systemctl".into(), "restart".into(), "{}".into()],
+            stop: vec!["/bin/systemctl".into(), "stop".into(), "{}".into()],
+            enable: vec!["/bin/systemctl".into(), "enable".into(), "{}".into()],
+            disable: vec!["/bin/systemctl".into(), "disable".into(), "{}".into()],
+            is_active: vec!["/bin/systemctl".into(), "is-active".into(), "{}".into()],
+        }
+    }
+}
+
+impl Default for SystemConfig {
+    fn default() -> Self {
+        Self {
+            init: InitConfig::default(),
+        }
+    }
+}
+
+impl SystemConfig {
+    pub fn try_new(config_root: PathBuf) -> Result<Self, SystemServiceError> {
+        let config_path = config_root.join(SERVICE_CONFIG_FILE);
+        let config_path_str = config_path.to_str().unwrap(); // always config_path exists
```

Comment:
The `unwrap()` is not related to the file existence but to whether or not the path is UTF8.

That's okay here since the path is not a user input ... unless we resurrect the idea of running tedge as a non-root user.

```suggestion
        let config_path_str = config_path.to_str().unwrap_or(SERVICE_CONFIG_FILE); 
```